Yoast/YoastSEO.js
# Add worker scheduler tests

## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Fixed stop polling in the scheduler by:
  - `tick` saves the setTimeout handle to `_pollHandle`.
  - `stopPolling` sets `_started` to `false`.
* Added tests for worker scheduler and worker scheduler task.

## Test instructions

This PR can be tested by following these steps:

* Check out the branch: `git checkout 1764-add-worker-scheduler-tests`.
* Install packages: `yarn`.
* Run the tests: `yarn test`.
* Verify the tests succeed.
* Open the coverage file for the scheduler: `/coverage/lcov-report/src/worker/scheduler/index.html`.
* Verify that the coverage is 100%!

Fixes #1764
